src/Pkg.jl

```diff
@@ -103,24 +103,32 @@ Add a package to the current project. This package will be available by using th
 a package, also inside that package.
 
 ## Resolution Tiers
-`Pkg` resolves the set of packages in your environment using a tiered algorithm.
-The `preserve` keyword argument allows you to key into a specific tier in the resolve algorithm.
-The following table describes the argument values for `preserve` (in order of strictness):
+`Pkg` resolves the set of packages in your environment using an algorithm that operates in one of four
+tiers, or modes. Which of these is used can be controlled by the `preserve` keyword argument,
+which should take one of the following values (in order of decreasing strictness):
 
 | Value             | Description                                                                         |
 |:------------------|:------------------------------------------------------------------------------------|
 | `PRESERVE_ALL`    | Preserve the state of all existing dependencies (including recursive dependencies)  |
-| `PRESERVE_DIRECT` | Preserve the state of all existing direct dependencies                              |
+| `PRESERVE_DIRECT` | Preserve the state of all existing direct dependencies. Error if this is impossible.|
 | `PRESERVE_SEMVER` | Preserve semver-compatible versions of direct dependencies                          |
 | `PRESERVE_NONE`   | Do not attempt to preserve any version information                                  |
-| `PRESERVE_TIERED` | Use the tier which will preserve the most version information (this is the default) |
+| `PRESERVE_TIERED` | Select from the above tiers the one which preserves the most version information, while still alowing a solution. (This is the default.) |
```

Comment:
"alowing" -> "allowing"

Adding the "while still allowing" feels redundant.

Reply:
Indeed, will fix.

Reply:
And perhaps I mis-read you. But I think "while allowing a solution" is important, as otherwise this sentence implies that it would always choose the first. It preserves the most, conditional on working.